### DISCUSSION TITLE
Add Windows and MAC agent support and RDP support.

### DISCUSSION THREAD

**Discussion by freedbygrace** [shellhub-io/shellhub#4198] [Ideas]
Windows has the ability to run a SSH server for many years so being able to manage those devices would be awesome! RDP, or VNC support would be awesome as well.

Being able to have other storage providers for remote recording storage or some kind of integration with SFTPGo. If not, this sort of stuff can be configured with RClone or something.

Simplify product deployment also. A native docker compose with the ability to specify an initial admin so that automated deployment can take place. As it stands, it seems that each instance needs some touch before being able to get it up and going?

This would launch this product into the stratosphere!

**Comment by heiytor** [comment]
Hi, @freedbygrace,

1. Native Windows and MAC support is not currently in our roadmap. However, if you're interested in contributing, feel free to check out and contribute to the agent codebase (`pkg/agent`). That said, the agent can run on Windows via WSL—just be sure to configure a [preferred hostname](https://github.com/shellhub-io/shellhub/blob/master/pkg/agent/agent.go#L107).

2. To clarify, are you asking for a custom provider at the server level, or would you prefer an option for each namespace to configure its own provider (for example, using a webhook)? Session recording is a feature exclusive to our **cloud** and **enterprise** instances. We plan to implement this feature soon by enabling change the provider at the server level, but it won't be available in the **community** instance. 

3. I'm not entirely sure I understand your third point. Are you requesting an environment that sets a user and namespace on startup? Also, it's possible to configure an initial admin using the `bin/cli` binary. After initializing the server, you can run the following commands in the `shellhub` directory:

```sh
$ ./bin/cli user create john_doe my_password john.doe@test.com
$ ./bin/cli namespace create my_namespace john_doe [a random uuid]
```

Please note that server admins with the ability to manage all users and namespaces are only available in the **enterprise** instance.

Lastly, could you explain what you meant by saying these changes would "launch this product into the stratosphere"?
